crates/context/config/src/client/env/proxy/types/starknet.rs

```diff
@@ -0,0 +1,341 @@
+use starknet::core::codec::{Decode, Encode};
+use starknet::core::types::{Felt, U256};
+
+use crate::repr::{Repr, ReprBytes, ReprTransmute};
+use crate::types::{ContextIdentity, ProposalId, SignerId};
+use crate::{
+    Proposal, ProposalAction, ProposalApprovalWithSigner, ProposalWithApprovals, ProxyMutateRequest,
+};
+
+#[derive(Debug, Encode, Decode)]
+pub struct FeltPair {
+    pub high: Felt,
+    pub low: Felt,
+}
+
+#[derive(Debug, Encode, Decode)]
+pub struct StarknetIdentity(pub FeltPair);
+
+#[derive(Debug, Encode, Decode)]
+pub struct StarknetProposalId(pub FeltPair);
+
+#[derive(Debug, Encode, Decode)]
+pub struct StarknetU256(pub FeltPair);
+
+#[derive(Debug, Encode, Decode)]
+pub struct StarknetProposal {
+    pub proposal_id: StarknetProposalId,
+    pub author_id: StarknetIdentity,
+    pub actions: StarknetProposalActionWithArgs,
+}
+
+#[derive(Debug, Encode, Decode)]
+pub struct StarknetConfirmationRequest {
+    pub proposal_id: StarknetProposalId,
+    pub signer_id: StarknetIdentity,
+    pub added_timestamp: Felt,
+}
+
+#[derive(Debug, Encode, Decode)]
+pub struct StarknetProxyMutateRequest {
+    pub signer_id: StarknetIdentity,
+    pub kind: StarknetProxyMutateRequestKind,
+}
+
+#[derive(Debug, Encode, Decode)]
+pub enum StarknetProxyMutateRequestKind {
+    Propose(StarknetProposal),
+    Approve(StarknetConfirmationRequest),
+}
+
+#[derive(Debug, Encode, Decode)]
+pub enum StarknetProposalActionWithArgs {
+    ExternalFunctionCall(Felt, Felt, Vec<Felt>),
+    Transfer(Felt, StarknetU256),
+    SetNumApprovals(Felt),
+    SetActiveProposalsLimit(Felt),
+    SetContextValue(Vec<Felt>, Vec<Felt>),
+}
+
+#[derive(Debug, Encode, Decode)]
+pub struct StarknetSignedRequest {
+    pub payload: Vec<Felt>,
+    pub signature_r: Felt,
+    pub signature_s: Felt,
+}
+
+#[derive(Debug, Decode)]
+pub struct StarknetProposalWithApprovals {
+    pub proposal_id: StarknetProposalId,
+    pub num_approvals: Felt,
+}
+
+#[derive(Debug, Decode)]
+pub struct StarknetApprovers {
+    pub approvers: Vec<StarknetIdentity>,
+}
+
+#[derive(Debug, Decode)]
+pub struct StarknetProposals {
+    pub proposals: Vec<StarknetProposal>,
+}
+
+impl From<StarknetProposals> for Vec<Proposal> {
+    fn from(value: StarknetProposals) -> Self {
+        value.proposals.into_iter().map(Into::into).collect()
+    }
+}
+
+// Conversions for StarknetIdentity
+impl From<Repr<SignerId>> for StarknetIdentity {
```

Comment:
`From<Repr<T>>` shouldn't be necessary, looks like we can benefit from `FeltPair` here